docs/pluggable-discovery-specification.md

```diff
@@ -0,0 +1,368 @@
+  client/IDE.
+- if the discovery cannot go into compatibility mode, it will report the protocol version supported (even if greater
+  than the version supported by the client/IDE) and the client/IDE may decide to terminate the discovery or produce an
+  error/warning.
+
+#### START command
+
+The `START` command initializes and start the discovery internal subroutines. This command must be called before `LIST`
+or `START_SYNC`. The response to the start command is:
+
+```JSON
+{
+  "eventType": "start",
+  "message": "OK"
+}
+```
+
+If the discovery could not start, for any reason, it must report the error with:
+
+```JSON
+{
+  "eventType": "start",
+  "error": true,
+  "message": "Permission error"
+}
+```
+
+The `error` field must be set to `true` and the `message` field should contain a description of the error.
+
+#### STOP command
+
+The `STOP` command stops the discovery internal subroutines and possibly free the internally used resources. This
+command should be called if the client wants to pause the discovery for a while. The response to the command is:
+
+```JSON
+{
+  "eventType": "stop",
+  "message": "OK"
+}
+```
+
+If an error occurs:
+
+```JSON
+{
+  "eventType": "stop",
+  "error": true,
+  "message": "Resource busy"
+}
+```
+
+The `error` field must be set to `true` and the `message` field should contain a description of the error.
+
+#### QUIT command
+
+The `QUIT` command terminates the discovery. The response to `QUIT` is:
+
+```JSON
+{
+  "eventType": "quit",
+  "message": "OK"
+}
+```
+
+after this output the discovery exits. This command is supposed to always succeed.
+
+#### LIST command
+
+The `LIST` command executes an enumeration of the ports and returns a list of the available ports at the moment of the
+call. The format of the response is the following:
+
+```
+{
+  "eventType": "list",
+  "ports": [
+    {
+      "address":       <-- THE ADDRESS OF THE PORT
+      "label":         <-- HOW THE PORT IS DISPLAYED ON THE GUI
+      "protocol":      <-- THE PROTOCOL USED BY THE BOARD
+      "protocolLabel": <-- HOW THE PROTOCOL IS DISPLAYED ON THE GUI
+      "properties": {
+                       <-- A LIST OF PROPERTIES OF THE PORT
+      }
+    }, {
+      ...              <-- OTHER PORTS...
+    }
+  ]
+}
+```
+
+The `ports` field contains a list of the available ports.
+
+Each port has:
+
+- an `address` (for example `/dev/ttyACM0` for serial ports or `192.168.10.100` for network ports)
+
+- a `label` that is the human readable form of the `address` (it may be for example `ttyACM0` or
+  `SSH on 192.168.10.100`)
+
+- `protocol` is the protocol identifier (such as `serial` or `dfu` or `ssh`)
+
+- `protocolLabel` is the `protocol` in human readable form (for example `Serial port` or `DFU USB` or `Network (ssh)`)
+
+- `properties` is a list of key/value pairs that represent information relative to the specific port
+
+To make the above more clear let’s show an example with the `serial_discovery`:
```

Comment:
I don't understand what `serial_discovery` is. I don't find any other occurrences of this identifier anywhere else in the documentation or the RFC.

Reply:
I changed it, should be clearer now.